src/UnitTests/GitHub.App/Controllers/UIControllerTests.cs

```diff
@@ -308,37 +446,71 @@ public void AuthFlowWith2FA()
             {
+                const int testViewCount = 3;
+
                 var count = 0;
+                bool? success = null;
+
+                //Starting Clone flow
                 var flow = uiController.SelectFlow(UIControllerFlow.Clone);
+                uiController.ListenToCompletionState()
+                    .Subscribe(s =>
+                    {
+                        success = s;
+                        Assert.Equal(testViewCount, count);
+                        count++;
+                    });
                 flow.Subscribe(data =>
                 {
                     var uc = data.View;
                     switch (++count)
                     {
                         case 1:
+                            //Demonstrate view is ILoginControlViewModel
                             Assert.IsAssignableFrom<IViewFor<ILoginControlViewModel>>(uc);
+
+                            //Displaying the TwoFactorView
                             var vm = factory.CreateViewAndViewModel(GitHub.Exports.UIViewType.TwoFactor).ViewModel;
                             vm.IsShowing.Returns(true);
                             RaisePropertyChange(vm, "IsShowing");
+
                             break;
                         case 2:
+                            //Demonstrate view is ITwoFactorDialogViewModel
                             Assert.IsAssignableFrom<IViewFor<ITwoFactorDialogViewModel>>(uc);
-                            // login
+
+                            //Login
                             cons.Add(SetupConnection(provider, hosts, hosts.GitHubHost));
-                            // continue by triggering done on login view
+                            
+                            //Continue by triggering done on login view
                             var v = factory.CreateViewAndViewModel(GitHub.Exports.UIViewType.Login).View;
                             TriggerDone(v);
+
                             break;
                         case 3:
+                            //Demonstrate view is IRepositoryCloneViewModel
                             Assert.IsAssignableFrom<IViewFor<IRepositoryCloneViewModel>>(uc);
+
+                            //Cancelling Clone flow
                             TriggerCancel(uc);
+
+                            break;
+
+                        default:
+                            Assert.True(false, "Received more views than expected");
                             break;
                     }
+                }, () =>
+                {
+                    Assert.Equal(testViewCount + 1, count);
+                    count++;
                 });
 
                 uiController.Start(null);
-                Assert.Equal(3, count);
+                Assert.Equal(testViewCount + 2, count);
                 Assert.True(uiController.IsStopped);
+                Assert.True(success.HasValue);
+                Assert.True(success);
```

Comment:
`Assert.True(success);` bothers me here. If the last action performed was Cancel on the clone flow, shouldn't this value be False?